streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -218,13 +214,9 @@ public boolean equals(final Object o) {
             Objects.equals(headers, that.headers);
     }
 
-    /**
-     * Equality is implemented in support of tests, *not* for use in Hash collections, since this class is mutable.
-     */
-    @Deprecated
     @Override
     public int hashCode() {
-        throw new UnsupportedOperationException("ProcessorRecordContext is unsafe for use in Hash collections");
+        return Objects.hash(timestamp, offset, topic, partition, headers);
```

Comment:
This class is unsafe for hashing, because Headers is mutable. Do you need to store it in a hash-collection?

Reply:
Ah. I forgot about headers. Don't think we need it. Will revert it.